### PR TITLE
Quick Nav Refactor and Clean Up, 2 Extra Tabs, Config Version 3, Config Data Fixer Refactor, Debug Dump Hovered Item Tool

### PR DESCRIPTION
**Pull request by VeritasDL**
added 2 extra quick nav tabs and tweaked rendering to accommodate the new tabs

### CODE REVIEW

**Comment by kevinthegreat1** [comment]
Thank you for your pull request. This is a great pr but there seems to be some slight disagreement due to the reordering of this feature. In addition with the large volume of pr's we have right now, please allow some time for review. Your pr has been put in bleeding edge, and you can download the newest bleeding edge build [here](https://nightly.link/SkyblockerMod/Skyblocker/workflows/beta/bleeding-edge).

**Comment by VeritasDL** [comment]
I think I've added a working config for disabling the extra tabs, I'm sure the codes straight spaghetti but ¯\_(ツ)_/¯

**Comment by kevinthegreat1** [comment]
I said this on discord but just so the information is on GitHub too:

Since this restructures the config, we're waiting until 1.20.5 is merged and will include this when we overhaul the config, to not mess up people's quick nav configs, which will likely be the next update.

Marking as wip for now.

**Comment by kevinthegreat1** [comment]
Conflicts with datafixer pr. Will need to wait for #701 to be merged.

**Comment by kevinthegreat1** [comment]
Hey can you please do a quick rebase? Then I can add the config shifting and data fixer stuff, unless you want to do it.

**Comment by kevinthegreat1** [comment]
WIP. Defaults and data fixer needs to be added.

**Comment by kevinthegreat1** [comment]
Basically rewrote quick nav.
Screenshot 2024-05-15 at 16 48 29Screenshot 2024-05-15 at 16 48 37

**Review by kevinthegreat1 — DISMISSED** [review]

**Review by kevinthegreat1 — APPROVED** [review]

File: src/main/java/de/hysky/skyblocker/skyblock/quicknav/QuickNav.java [hunk 1]
```diff
@@ -51,29 +37,28 @@ public static List<QuickNavButton> init(String screenTitle) {
...
         }
         return buttons;
     }
 
     private static QuickNavButton parseButton(QuickNavigationConfig.QuickNavItem buttonInfo, String screenTitle, int id) throws CommandSyntaxException {
-        QuickNavigationConfig.ItemData itemData = buttonInfo.item;
-        ItemStack stack = ItemStackComponentizationFixer.fromComponentsString(itemData.id, Math.clamp(itemData.count, 1, 99), itemData.components);
-        boolean uiTitleMatches = false;
+        QuickNavigationConfig.ItemData itemData = buttonInfo.itemData;
+        ItemStack stack = itemData != null && itemData.item != null && itemData.components != null ? ItemStackComponentizationFixer.fromComponentsString(itemData.item.toString(), Math.clamp(itemData.count, 1, 99), itemData.components) : new ItemStack(Items.BEDROCK);
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/quicknav/QuickNav.java:50]: The default should be barrier

**Review by AzureAaron — DISMISSED** [review]

**Review by kevinthegreat1 — APPROVED** [review]

**Pull request merged**